pkg/config/crd/store_test.go

```diff
@@ -61,10 +61,15 @@ func createFakeDiscovery(*rest.Config) (discovery.DiscoveryInterface, error) {
 type dummyListerWatcherBuilder struct {
 	mu       sync.RWMutex
 	data     map[store.Key]*unstructured.Unstructured
-	watchers map[string]*watch.FakeWatcher
+	watchers map[string]*watch.RaceFreeFakeWatcher
 }
 
 func (d *dummyListerWatcherBuilder) build(res metav1.APIResource) cache.ListerWatcher {
+	w := watch.NewRaceFreeFake()
+	d.mu.Lock()
```

Comment:
Any reason to not do simple lock-> edit -> unlock..
defer will keep the lock for longer that needed (tll the entire mtd ends)

d.mu.Lock()
d.watchers[res.Kind] = w
d.mu.Unlock()
?

Same for the defer Unlocks below

Reply:
Changed here and below `RLock`.  I think defer is better for `put` and `delete`, since it can early-exit and use of defer simplifies the code.